eng/native/configurecompiler.cmake

```diff
@@ -262,18 +266,22 @@ endif(CLR_CMAKE_HOST_WIN32)
 # Architecture specific files folder name
 if (CLR_CMAKE_TARGET_ARCH_AMD64)
     set(ARCH_SOURCES_DIR amd64)
-    add_definitions(-DTARGET_AMD64)
-    add_definitions(-DTARGET_64BIT)
+    set(ARCH_TARGET_NAME x64)
+    add_compile_definitions($<$<NOT:$<BOOL:$<TARGET_PROPERTY:IGNORE_DEFAULT_TARGET_ARCH>>>:TARGET_AMD64>)
```

Comment:
This just means - add the definition only if the target being compiled doesn't have the `IGNORE_DEFAULT_TARGET_ARCH `property set. 
But I am not happy with this way of solving the actual problem. This file is meant to  define general properties for all targets being built in the runtime repo (in coreclr, libs, installer) and possibly other repos that include this file (I think e.g. diagnostics repo does that too). What we do here is special to coreclr. It also seems error prone - it will be easy to forget / hard to figure out when adding a new define in the future where to use this conditional adding of target property and when to not to use it and we now have the same defines created at two places (here and in the `src/coreclr/clrdefinitions.cmake` for the case we use the `IGNORE_DEFAULT_TARGET_ARCH` property.
Since we already had a build of some cross components (including e.g. x64 hosted jit for x86 / arm) before your change, so I wonder why we cannot use the same mechanism using the `TARGET_*` / `HOST_*` defines.